benchmarks/polys.py

```diff
@@ -1,17 +1,124 @@
-import sympy
+from sympy import symbols, prod, prem
+from sympy.polys import ZZ, Poly
+
 
 class TimePolyManyGens:
     """Time using a Poly with many generators"""
 
     params = [1, 10, 100, 500]
 
     def setup(self, n):
-        self.xs = sympy.symbols('x:{}'.format(n))
+        self.xs = symbols('x:{}'.format(n))
         self.x = self.xs[n // 2]
-        self.px = sympy.Poly(self.x, self.xs)
+        self.px = Poly(self.x, self.xs)
 
     def time_create_poly(self, n):
-        sympy.Poly(self.x, self.xs)
+        Poly(self.x, self.xs)
 
     def time_is_linear(self, n):
         self.px.is_linear
+
+
+x, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10 = symbols("x, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10")
+y = [y1, y2, y3, y4, y5, y6, y7, y8, y9, y10]
+R = ZZ[x, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10]
```

Comment:
The variables like `y1, y2, ...` are not needed:
```
x, *y = symbols("x, y:10")
R = ZZ[x, *y]
```
These should not be global variables though so move them into the `generate_` methods.

Probably the best thing here is to do like the PyDy examples and have a superclass that can generate examples and subclasses that time specific methods with specific parameter values.

Reply:
```
x, *y = symbols("x, y:10")
R = ZZ[x, *y]
```
This syntax is only valid in Python 3.11 not in 3.10. For running benchmarks I need python 3.10.


```python

Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import *     
>>> x, *y = symbols("x, y:10")
>>> R = ZZ[x, *y]
  File "<stdin>", line 1
    R = ZZ[x, *y]
              ^
SyntaxError: invalid syntax

```